openssl/src/ssl/mod.rs

```diff
@@ -652,6 +654,17 @@ impl SslContextBuilder {
         }
     }
 
+    /// Sets a custom X509Store for verifying peer certificates
+    #[cfg(any(all(feature = "v102", ossl102), all(feature = "v110", ossl110)))]
+    pub fn set_verify_cert_store(&mut self, cert_store: X509Store) -> Result<(), ErrorStack> {
+        unsafe {
+            // set0 will free, set1 increments, and then requires a free
+            let ptr = cert_store.as_ptr();
+            mem::forget(cert_store);
```

Comment:
Yeah, I wasn't sure about that. It's not documented here: 

https://www.openssl.org/docs/man1.1.0/ssl/SSL_CTX_ctrl.html

And here: https://www.openssl.org/docs/man1.1.0/ssl/SSL_CTX_set0_verify_cert_store.html
the docs state:

```
...
Those containing a 0 do not increment reference counts and the supplied store MUST NOT be freed after the operation.
```

which led me to wanting to forget it prior to the call. It doesn't make an exception for the error case. What I'll do, is capture the return, then forget, then return the result of the call, sound good. you're right that this seems safer.